### PR TITLE
Support dotty

### PR DESCRIPTION
**Pull request by cquiroz**
Adds dotty support: Some details:

* scala native support dropped
* scala 2.11 support has been dropped
* documentation in process of being ported to mdoc
* tests can't be run on scala.js due to scalatest lack of support

### CODE REVIEW

File: shared/src/main/scala/squants/SVector.scala
Comment (cquiroz) [inline shared/src/main/scala/squants/SVector.scala:83]: This fails to compile in dotty. I'm not sure if what I'm doing is idiomatic

Comment (larsrh) [reply shared/src/main/scala/squants/SVector.scala:83]: Looks good to me.

File: shared/src/test/scala/squants/experimental/ExperimentalSpec.scala
Comment (larsrh) [inline shared/src/test/scala/squants/experimental/ExperimentalSpec.scala:20]: Why are these tests commented out?

File: shared/src/test/scala/squants/experimental/ExperimentalSpec.scala
Comment (larsrh) [inline shared/src/test/scala/squants/experimental/ExperimentalSpec.scala:61]: This should not be necessary, I think

File: shared/src/test/scala/squants/experimental/ExperimentalSpec.scala
Comment (cquiroz) [reply shared/src/test/scala/squants/experimental/ExperimentalSpec.scala:20]: I had troubles compiling them in dotty, I tried a few variants and gave up, especially considering this is "experimental" code

**Comment by AtelierFox** [comment]
Would love to help if something's still throwing errors, as I want to move a personal project to Scala 3 and this seems like it's almost there. Is there anything on this pull that still isn't working or needs fixing?

**Comment by cquiroz** [comment]
I'll clean up the comments by @larsrh and it just needs to get an approval. We can make a release after merging

**Comment by cquiroz** [comment]
@larsrh Do you think I could get merge this one?

File: .github/workflows/scala.yml
Comment (larsrh) [inline .github/workflows/scala.yml:19]: `squantsNative` should be readded

File: project/Build.scala
Comment (larsrh) [inline project/Build.scala:12]: Bump to `RC2` to get a suitable ScalaTest

File: project/Build.scala
Comment (larsrh) [inline project/Build.scala:62]: Remove both

**Review by larsrh — CHANGES_REQUESTED** [review]
The [issue with some tests](https://github.com/typelevel/squants/pull/454#discussion_r591830796) is still open; I recommend either moving them to a Scala 2.x subfolder (I think sbt should do that by default now) or remove the experimental code altogether. Keeping them in but commented is not great.

**Comment by cquiroz** [comment]
I deleted the experimental code, I wanted to keep it but the build started to get very complicated 

let me know if this needs more changes

**Review by larsrh — APPROVED** [review]
LGTM!

File: build.sbt
Comment (julienrf) [inline build.sbt:36]: By enabling the `MdocPlugin` in the library, this adds the `mdoc` artifacts as dependencies of squants itself. I don’t think this should be the case.

Screenshot from 2021-05-06 16-37-12

Instead, the `MdocPlugin` should be enabled to a separate “documentation” project.

Comment (julienrf) [reply build.sbt:36]: I’ve tried to address this problem but I faced other issues. The most important one is that the current README.md does not compile with mdoc.

Comment (julienrf) [reply build.sbt:36]: I’ve created #467 to fix this problem.

**Pull request merged**
